csharp/test/Apache.Arrow.Tests/SchemaBuilderTests.cs

```diff
@@ -16,12 +16,36 @@
 using Apache.Arrow.Types;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace Apache.Arrow.Tests
 {
     public class SchemaBuilderTests
     {
+        public class SchemaComparer        {
+            public static bool Equals(Schema s1, Schema s2)
+            {
+                if (s2 == null || s1 == null || s1.HasMetadata != s2.HasMetadata || s1.Fields.Count != s2.Fields.Count)
+                {
+                    return false;
+                }
+
+                if (!s1.Fields.Keys.All(k => s2.Fields.ContainsKey(k) && FieldComparer.Equals(s1.Fields[k], s2.Fields[k])) ||
```

Comment:
Question for the crowd: Do the Fields need to be in the same order to be considered "equal" schemas?

For example, if I have:

```
Name
Age
Address
```

Is that "equal" to:

```
Address
Age
Name
```
?

Reply:
Right, I was going to ask the same question. I'm ignoring order here because it was simpler. Schema only exposes a dictionary, so it has no real sense of order in the C# implementation. The cpp implementation in type.cc on the other hand suggests that the order is important. Let's see what the others say. In any case, I don't think this is an important issue (yet)

Reply:
Schema exposes:

https://github.com/apache/arrow/blob/6fb4cd0e00fb17aaf80113280ee642494ff3296e/csharp/src/Apache.Arrow/Schema.cs#L50-L53

and 

https://github.com/apache/arrow/blob/6fb4cd0e00fb17aaf80113280ee642494ff3296e/csharp/src/Apache.Arrow/Schema.cs#L58-L65

Which shows that it does have a sense of order.